java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -78,6 +69,38 @@ public JDBCSink(JDBCSinkConfig config, TableSchema tableSchema) {
                             String.format(ERROR_REPORT_TEMPLATE, e.getSQLState(), e.getMessage()))
                     .asRuntimeException();
         }
+
+        // Upsert sink must have primary key
+        if (config.isUpsertSink() && pkColumnNames.isEmpty()) {
```

Comment:
We shall add this logic to JdbcSinkFactory.validate to report error on the create sink statement.